src/components/TextInput.tsx

```diff
@@ -1,63 +1,53 @@
-/**
- * @author Angela Guo <aguo921@aucklanduni.ac.nz>
- */
-
 "use client";
 
-import { useState } from "react";
+import type { InputHTMLAttributes } from "react";
 import { twMerge } from "tailwind-merge";
 
 import { cn } from "@/lib/utils";
 
-type TextInputProps = {
-  label?: string;
-  value: string;
+interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
+  label: string;
   type: string;
-  isError: boolean;
-  backgroundColor?: string;
-  className?: string;
-  onChange: (value: string) => void;
-};
+  isError?: boolean;
+  OnChange?: (value: string) => void;
```

Comment:
you don't need to add this as InputHTMLAttributes< HTMLInputElement> already has onchange